src/Service/Api/DataApi.php

```diff
@@ -278,27 +204,26 @@ public function getSegments() {
    *   The type of event, can be one of 'CAMPAIGN_ACTION', 'CAMPAIGN_CLICK_THROUGH',
    *   'CAMPAIGN_CONVERSION', or 'OTHER' (default).
    *
-   * @throws AcquiaLiftDataConnectorException
+   * @throws DataApiException
    */
   public function saveEvent($event_name, $event_type = 'OTHER') {
     // First get our Authorization header.
-    $headers = array('Accept' => 'application/json');
+    $headers = ['Accept' => 'application/json'];
     $url = $this->generateEndpoint('events/' . $event_name);
-    $auth_header = $this->getAuthHeader('PUT', $url, array('type' => $event_type), $headers);
-    $headers += array('Authorization' => $auth_header);
+    $auth_header = $this->getAuthHeader('PUT', $url, ['type' => $event_type], $headers);
+    $headers += ['Authorization' => $auth_header];
 
     $request = new Request('PUT', $url . '?type=' . $event_type, $headers);
-    $response = $this->httpClient()->send($request);
-    $vars = array('@eventname' => $event_name);
+    $response = $this->httpClient->send($request);
+    $vars = ['@eventname' => $event_name];
     $success_msg = SafeMarkup::format('The event @eventname has been saved to Acquia Lift', $vars);
     $fail_msg = SafeMarkup::format('Could not save event @eventname to Acquia Lift', $vars);
     if ($response->getStatusCode() == 200) {
       $this->logger->info($success_msg);
+      return;
     }
-    else {
-      $this->logger->error($fail_msg);
-      throw new AcquiaLiftDataConnectorException($fail_msg);
-    }
+    $this->logger->error($fail_msg);
+    throw new DataApiException($fail_msg);
   }
 
```

Comment:
In case you noticed there is no more "else" keyword in this code base - this approach is an alternative and I don't plan to actively advocate for this guideline at Acquia.

I do have an upcoming talk at my local group on "Object Calisthenics", which one guideline is to "avoid 'else' keyword as much as possible", and I was hoping to use this code base as a part of demo to show that you almost always code better if you avoid "else".